### PR TITLE
Adding new job completed event

### PR DESCRIPTION
**Pull request by raz-bn**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Adding job completed event.
Usefull for gathering more information about jobs running in the cluster only from the event log.

Example:
```
root@raz:/home/raz/kubernetes# cluster/kubectl.sh get events
LAST SEEN   TYPE     REASON                    OBJECT           MESSAGE
<unknown>   Normal   Scheduled                 pod/pi-hvvjp     Successfully assigned default/pi-hvvjp to 127.0.0.1
105s        Normal   Pulling                   pod/pi-hvvjp     Pulling image "perl"
22s         Normal   Pulled                    pod/pi-hvvjp     Successfully pulled image "perl"
16s         Normal   Created                   pod/pi-hvvjp     Created container pi
16s         Normal   Started                   pod/pi-hvvjp     Started container pi
107s        Normal   SuccessfulCreate          job/pi           Created pod: pi-hvvjp
8s          Normal   Completed                 job/pi           Job default/pi completed
```
**Which issue(s) this PR fixes**:
None.

**Special notes for your reviewer**:
None.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by raz-bn** [comment]
CLA fixed.


> 
> 
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please follow instructions at https://git.k8s.io/community/CLA.md#the-contributor-license-agreement to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment and we'll verify. Thanks.
> 
>     * If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address.  Check your existing CLA data and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
>     * If you signed the CLA as a corporation, please sign in with your organization's credentials at https://identity.linuxfoundation.org/projects/cncf to be authorized.
> 
>     * If you have done the above and are still having issues with the CLA being reported as unsigned, please log a ticket with the Linux Foundation Helpdesk: https://support.linuxfoundation.org/
> 
>     * Should you encounter any issues with the Linux Foundation Helpdesk, send a message to the backup e-mail support address at: [login-issues@jira.linuxfoundation.org](mailto:login-issues@jira.linuxfoundation.org)
> 
> 
> Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md). If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).

**Comment by raz-bn** [comment]
cla fixed.

File: pkg/controller/job/job_controller.go [hunk 1]
```diff
@@ -566,6 +566,7 @@ func (jm *JobController) syncJob(key string) (bool, error) {
 			job.Status.Conditions = append(job.Status.Conditions, newCondition(batch.JobComplete, "", ""))
 			now := metav1.Now()
 			job.Status.CompletionTime = &now
+			jm.recorder.Event(&job, v1.EventTypeNormal, "Completed", fmt.Sprintf("Job %s completed", key))
```
Comment (smarterclayton) [inline pkg/controller/job/job_controller.go]: Remove the job name from the event - the event object already has it (as a field)

**Comment by smarterclayton** [comment]
/lgtm
/approve

Thanks

**Pull request merged**